discussion/rust-for-embedded-systems.tex

```diff
@@ -0,0 +1,90 @@
+% !TEX root = ../main.tex
+
+\section{Rust for Embedded Systems} % (fold)
+\label{sec:disc:rust_for_arm}
+% \todo{Mention this in discussion. Is there a better way of solving this? Currently it fetches a release tarball of the language. It fetches more than it needs to for each library, but it's quicker/easier than using e.g. git to do it. It does however take a little while to compile all the dependencies on smaller machines.}
+
+Before we started on this thesis, we were already aware that there were a couple of other projects that were running {\rust} bare-metal on ARM \glspl{mcu}.
+We did, however, not know how easy it would be to get up and running with {\rust} on the EFM32.
+It took us less than one day to implement and execute a {\rust} blink-demo on the {\STK}.
+The {\rust} documentation was sufficient for figuring out how to define non-std {\rust} programs (i.e. the ones that are annotated with \texttt{\#[no\_std]}), and projects like \texttt{armboot} helped us build the project for the ARM architecture.
+{\rust}'s runtime is comparable to {\C}'s, because the startup routine in the two languages is the same, but {\rust}'s {\core} library is larger and provides more functionality than the {\C} standard library, which we consider to be an added bonus of {\rust}.
+
+\subsection{The Standard Library} % (fold)
+\label{sub:using_the_standard_library}
+
+% subsection using_the_standard_library (end)
+
+{\rust} has a rich standard library that offers a strong foundation for all the programs that want to depend on it.
+As we described in \autoref{ssub:rust:features}, the {\std} crate acts like an abstraction over all the different libraries in \gls{rsl} and the {\rustc} compiler is itself dependent on a couple of traits to be present in this module.
+It is easy enough to define programs that are not dependent on this {\std} library, and instead only dependent on {\core}, but we think the difference between {\std} and {\core} is too large.
```

Comment:
`on {\core}. We have found that for our platform the difference ...`
eller `on {\core}. We have found that for our platform the distinction between {\std} and {\core} is not put on a perfect level`